Design.md

```diff
@@ -32,71 +32,73 @@ Here are the IELE design decisions described from the perspective of differences
-* Following the data segment are one or more functions. A function begins with the CALLDEST or EXTCALLDEST instruction, and continues until the next CALLDEST or EXTCALLDEST instruction, or until the end of the contract. The contract is malformed if a CALLDEST or EXTCALLDEST instruction appears before the end of the data segment.
-* A CALLDEST instruction takes an immediate function label, and an immediate number representing the number of arguments to the function. This function is not externally exported. As a special case, function label zero is the initialization routine of the contract. It is an exception if a LOCALCALL instruction invokes this function. If this function is not declared, the contract is malformed.
-* A EXTCALLDEST instruction takes an immedate index in the data segment, and an immediate number representing the number of arguments to the function. It is malformed if the data segment index does not refer to a FUNCTION instruction. The FUNCTION instruction provides the external name of the function as its string value. This function is externally exported. The function label of this function is its data segment index.
-* The contract is malformed if multiple functions declare the same function label.
+* An external contract declaration simply declares the name of another contract. The contract under definition can only create accounts with deployed code being a copy of the contract itself or one of the contracts that have been declared as external (see section on contract creation for details). Each externally declared contract should have been defined in the same file and above the contract that externally declares it.
+* A function definition includes the function signature, the function body and whether or not the function is public. A function signature includes a function name (which is also represented as a global register, as it is visible across the contract) and names of formal arguments (which are represented as local registers, as they are only visible within the function body).
+* A public function can be called by other accounts, while a non-public one can only be called by other functions within the same contract.
+* A special function named `@init` is automatically added to the contract (and can be overridden by the programmer), and will be called when an account is created with this contract.
+* An account to which code has never been deployed contains an implicit public function `@deposit` which takes no arguments, returns no values, and does nothing. This function exists to allow accounts to receive payment even if they do not have a contract deployed to them. Note that a contract can forbid payments by refusing to declare the `@deposit` function, and explicitly raising an exception if any of its entry points are invoked with a balance transfer.
```

Comment:
I do not understand this.  If the deposit function does nothing, then how can the account receive payment?  Also, looks questionable to allow either a function or a contract in an account.  Or both?  Or a contract with a deposit function in it?

Reply:
I assume that calling `@deposit` in an account without code doesn't require passing any `gaslimit`. Do we force a `@deposit` function in a contract to do nothing (besides receiving balance), and allow it to be called with zero gas?

Reply:
The payment is independent of the code in the body of the function. If you see the syntax for account calls they look like: r1, r2 = call @FunName at Address ( Args ) send Value, gaslimit Gas. When doing an account call the amount in Value is transfered to the account as part of the message even if the called function does nothing. This is similar to EVM if I am not mistaken.
@bmmoore I think your understanding is correct.